src/web/app/pages-instructor/instructor-session-modal-page.component.ts

```diff
@@ -0,0 +1,125 @@
+import { Router } from '@angular/router';
+import { NgbModal, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
+import { forkJoin } from 'rxjs';
+import { FeedbackQuestionsService } from '../../services/feedback-questions.service';
+import { FeedbackSessionsService } from '../../services/feedback-sessions.service';
+import { HttpRequestService } from '../../services/http-request.service';
+import { NavigationService } from '../../services/navigation.service';
+import { StatusMessageService } from '../../services/status-message.service';
+import { StudentService } from '../../services/student.service';
+import {
+  FeedbackSessionSubmittedGiverSet,
+  Student, Students,
+} from '../../types/api-output';
+import {
+  ResendResultsLinkToStudentModalComponent,
+} from "../components/sessions-table/resend-results-link-to-student-modal/resend-results-link-to-student-modal.component"; // tslint:disable-line
+import {
+  SendRemindersToStudentModalComponent,
+} from '../components/sessions-table/send-reminders-to-student-modal/send-reminders-to-student-modal.component';
+import {
+  SessionsTableRowModel,
+} from '../components/sessions-table/sessions-table-model';
+import {
+  StudentListInfoTableRowModel,
+} from '../components/sessions-table/student-list-info-table/student-list-info-table-model';
+import { ErrorMessageOutput } from '../error-message-output';
+import {
+  InstructorSessionBasePageComponent,
+} from './instructor-session-base-page.component';
+
+/**
+ * The base page for session related page.
+ */
+export abstract class InstructorSessionModalPageComponent extends InstructorSessionBasePageComponent {
+
+  protected constructor(router: Router,
+                        httpRequestService: HttpRequestService,
+                        statusMessageService: StatusMessageService,
+                        navigationService: NavigationService,
+                        feedbackSessionsService: FeedbackSessionsService,
+                        feedbackQuestionsService: FeedbackQuestionsService,
+                        protected modalService: NgbModal,
+                        protected studentService: StudentService) {
```

Comment:
Shouldn't the indentation be twice the normal?

Reply:
They are automatically aligned by IntelliJ. Though it "violates" the coding standard we published, I am OK with the indentation for the constructor as it is more clearly showing the services used.